core/src/apps/cardano/staking_use_cases.py

```diff
@@ -0,0 +1,50 @@
+from trezor.messages import CardanoAddressType
+
+from apps.cardano.address import get_public_key_hash, path_to_staking_path
+
+if False:
+    from trezor.messages import CardanoAddressParametersType
+    from apps.cardano import seed
+
+
+"""
+Used as a helper when deciding what warnings we should
+display to the user during get_address and sign_tx depending
+on the type of address and its parameters.
+"""
+
+
+class StakingUseCases:
+    NO_STAKING = 0
+    SAME_ACCOUNT = 1
+    SAME_HASH = 2
+    DIFFERENT_ACCOUNT = 3
+    DIFFERENT_HASH = 4
+    POINTER_ADDRESS = 5
+    REWARD_ADDRESS = 6
+
+    @classmethod
+    def get(
+        cls, keychain: seed.Keychain, address_parameters: CardanoAddressParametersType
+    ) -> int:
+        address_type = address_parameters.address_type
+        if address_type == CardanoAddressType.BASE_ADDRESS:
+            if address_parameters.staking_key_path:
+                staking_path = path_to_staking_path(address_parameters.path)
```

Comment:
spending_account_staking_path would imho be more descriptive name

Reply:
Definitely. Changed.